src/main/java/com/example/demo/service/book/BookServiceImpl.java

```diff
@@ -20,10 +25,18 @@ public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
     private final BookSpecificationBuilder bookSpecificationBuilder;
+    private final CategoryRepository categoryRepository;
 
     @Override
     public BookDto save(CreateBookRequestDto requestDto) {
         Book book = bookMapper.toModel(requestDto);
+
+        Set<Category> categories = requestDto.getCategoryIds().stream()
+                .map(categoryId -> categoryRepository.findById(categoryId)
+                        .orElseThrow(() -> new RuntimeException("Category not found")))
```

Comment:
add parameter to message 
```suggestion
                        .orElseThrow(() -> new EntityNotFoundException("Category not found")))
```